tests/test_models/test_models.py

```diff
@@ -0,0 +1,212 @@
+from app.models import Bucketlist, Item, User
+from datetime import datetime
+from tests import BaseTestCase, db
+from time import sleep
+
+
+class BaseModelTest(BaseTestCase):
+    def setUp(self):
+        super(BaseModelTest, self).setUp()
+        self.bucketlist = Bucketlist("Cook")
+        db.session.add(self.bucketlist)
+        db.session.commit()
+
+
+class TestBucketList(BaseModelTest):
+
+    def test_creating_bucketlist(self):
+        """Tests successfully creating a bucketlist"""
+
+        # this test would conflict with bucketlist defined in the setup
+        # clear everythin before running it
+        db.drop_all()
+        db.create_all()
+
+        # Instantiating a bucketlist object
+        bucketlist = Bucketlist("Cook")
+        # save the object to database
+        db.session.add(bucketlist)
+        db.session.commit()
+
+        # asssert the attributes
+        self.assertEqual(bucketlist.id, 1)
+        self.assertEqual(bucketlist.name, "Cook")
+        self.assertEqual(bucketlist.created_by, None)
+        year = str(datetime.today().year)
+        self.assertIn(year, str(bucketlist.date_created))
+        self.assertIn(year, str(bucketlist.date_modified))
+        self.assertEqual(len(bucketlist.items), 0)
+
+        # test querying bucketlists
+        bucketlist_query = Bucketlist.query.all()
+        self.assertIn("<Bucketlist 'Cook'>", str(bucketlist_query))
+        self.assertFalse("<Bucketlist 'Random'>" in str(bucketlist_query))
+
+    def test_creating_bucketlist_with_a_missing_name(self):
+        """Tests successfully creating a bucketlist"""
+
+        # this test would conflict with bucketlist defined in the setup
+        # clear everythin before running it
+        db.drop_all()
+        db.create_all()
```

Comment:
To avoid this in every test function, this class can have its own setup method that calls these two functions.